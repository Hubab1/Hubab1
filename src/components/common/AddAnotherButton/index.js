import React from 'react';
import PropTypes from 'prop-types';

import AddCircle from '@material-ui/icons/AddCircle';

const AddAnotherButton = (props) => (
    <span
        role="button"
        style={{
            color: '#2B44FF',
            display: 'inline-block',
            cursor: 'pointer'
        }}
        onClick={props.onClick}
    >
        <span style={{fontSize: props.fontSize}}>
            <AddCircle style={{verticalAlign: 'middle', fontSize: props.fontSize + 5}} /> <span style={{verticalAlign: 'middle'}}>Add Another {props.thing}</span>
        </span>
    </span>
);

AddAnotherButton.propTypes = {
    fontSize: PropTypes.number
}

AddAnotherButton.defaultProps = {
    fontSize: 13
}

export default AddAnotherButton;