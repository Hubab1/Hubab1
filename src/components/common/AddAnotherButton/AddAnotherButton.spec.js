import React from 'react';
import renderer from 'react-test-renderer';

import AddAnotherButton from './index';

let defaultProps;

beforeEach(() => {
    defaultProps = {
        theme: {
            palette: {
                primary: {
                    main: '#FFFFFF'
                }
            }
        },
        fontSize: 15,
        thing: 'Pet'
    };
});

describe('AddAnotherButton', function() {
    it('Matches snapshot', function() {
        expect(
            renderer.create(<AddAnotherButton {...defaultProps}/>)
        ).toMatchSnapshot();
    });
});
