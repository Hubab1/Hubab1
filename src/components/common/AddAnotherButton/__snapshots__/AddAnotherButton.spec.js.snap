// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAnotherButton Matches snapshot 1`] = `
<span
  role="button"
  style={
    Object {
      "color": "#2B44FF",
      "cursor": "pointer",
      "display": "inline-block",
    }
  }
>
  <span
    style={
      Object {
        "fontSize": 15,
      }
    }
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      style={
        Object {
          "fontSize": 20,
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
      />
    </svg>
     
    <span
      style={
        Object {
          "verticalAlign": "middle",
        }
      }
    >
      Add Another Pet
    </span>
  </span>
</span>
`;
